cmake/modules/CUDA.cmake

```diff
@@ -63,10 +70,14 @@ set(CUDA_gpu_detect_output "")
       # nvcc outputs text containing line breaks when building with MSVC.
       # The line below prevents CMake from inserting a variable with line
       # breaks in the cache
-      message(STATUS "Found CUDA arch ${__nvcc_out}")
+      message(STATUS "Found GPU arch ${__nvcc_out}")
       string(REGEX MATCH "([1-9].[0-9])" __nvcc_out "${__nvcc_out}")
-      string(REPLACE "2.1" "2.1(2.0)" __nvcc_out "${__nvcc_out}")
-      set(CUDA_gpu_detect_output ${__nvcc_out} CACHE INTERNAL "Returned GPU architetures from mshadow_detect_gpus tool" FORCE)
+      if(__nvcc_out VERSION_LESS "3.5")
+        # drop support for cc < 3.5 and build for all known archs.
+        message(WARNING "GPU arch is too old to support.")
```

Comment:
Recommend changing the error message to "GPU arch less than 3.5 is not supported."

Reply:
Changed accordingly.